src/ExtractTileOperations.cpp

```diff
@@ -39,12 +39,33 @@ Type amx_op_type_result_type(AMXOpType op_ty) {
     }
 }
 
+int amx_op_type_size(AMXOpType op_ty) {
+    switch (op_ty) {
+    case AMXOpType::Int8:
+        return 1;
+    case AMXOpType::Bfloat16:
+        return 2;
+    default:
+        internal_error << "Unexpected";
+        return -1;
+    }
+}
+
 const auto wild_i32 = Variable::make(Int(32), "*");
 const auto wild_i32x = Variable::make(Int(32, 0), "*");
 
 Tile<1> get_1d_tile_index(const Expr &e) {
     if (const auto *r1 = e.as<Ramp>()) {
-        return {true, r1->base, {r1->stride}, {r1->lanes}};
```

Comment:
Do we want some logic akin to HexagonOptimize's `apply_commutative_patterns`, here and below?

Reply:
would that be something like an `expr_match` that also checks the commutative?

Reply:
Yes, althought I believe `apply_commutative_patterns` only checks one level of commutation. I'm just wondering if we should be checking for all of these:
```
((wild_i32 * wild_i32) + wild_i32) * wild_i32
wild_i32 * ((wild_i32 * wild_i32) + wild_i32)
(wild_i32 + (wild_i32 * wild_i32)) * wild_i32
wild_i32 * (wild_i32 + (wild_i32 * wild_i32))
```

Reply:
That would make sense, to do that I feel like it requires to implement a new `expr_match` which applies those rules?

Reply:
I've played around a bit with extending the `expr_match` to support commutative operations where relevant, but I think for now I've also settled on the map approach.